source/pidomus_checkpoint_restart.cc

```diff
@@ -0,0 +1,411 @@
+              (uint32_t)oss.str().length(), /* size of block */
+              (uint32_t)oss.str().length(), /* size of last block */
+              (uint32_t)compressed_data_length
+            }; /* list of compressed sizes of blocks */
+
+        std::ofstream f ((snap_prefix + "restart.resume.z").c_str());
+        f.write((const char *)compression_header, 4 * sizeof(compression_header[0]));
+        f.write((char *)&compressed_data[0], compressed_data_length);
+      }
+#else
+    AssertThrow (false,
+                 ExcMessage ("You need to have deal.II configured with the 'libz' "
+                             "option to support checkpoint/restart, but deal.II "
+                             "did not detect its presence when you called 'cmake'."));
+#endif
+
+  }
+  pcout << "*** Snapshot created!" << std::endl << std::endl;
+}
+
+
+template <int dim, int spacedim, typename LAC>
+void
+piDoMUS<dim,spacedim,LAC>::
+save_solutions_and_triangulation(const LADealII::VectorType &y,
+                                 const LADealII::VectorType &y_dot,
+                                 const LADealII::VectorType &y_expl,
+                                 const LADealII::VectorType &,
+                                 const LADealII::VectorType &) const
+{
+  std::string sy = snap_prefix+"y.bin";
+  std::string sdot = snap_prefix+"y_dot.bin";
+  std::string sexpl = snap_prefix+"y_expl.bin";
+  const char *file_y = sy.c_str();
+  const char *file_y_dot = sdot.c_str();
+  const char *file_y_expl = sexpl.c_str();
+
+  if (file_exists(file_y))
+    copy_file(file_y,snap_prefix+"y.bin.old");
+  if (file_exists(file_y_dot))
+    copy_file(file_y_dot,snap_prefix+"y_dot.bin.old");
+  if (file_exists(file_y_expl))
+    copy_file(file_y_expl,snap_prefix+"y_expl.bin.old");
+
+  std::ofstream out_y (file_y);
+  std::ofstream out_y_dot (file_y_dot);
+  std::ofstream out_y_expl (file_y_expl);
+
+  y.block_write(out_y);
+  out_y.close();
+
+  y_dot.block_write(out_y_dot);
+  out_y_dot.close();
+
+  y_expl.block_write(out_y_expl);
+  out_y_expl.close();
+
+  triangulation->save ((snap_prefix + "restart.mesh").c_str());
+}
+
+template <int dim, int spacedim, typename LAC>
+void
+piDoMUS<dim,spacedim,LAC>::
+save_solutions_and_triangulation(const LATrilinos::VectorType &,
+                                 const LATrilinos::VectorType &,
+                                 const LATrilinos::VectorType &y_expl,
+                                 const LATrilinos::VectorType &y,
+                                 const LATrilinos::VectorType &y_dot) const
+{
+  std::vector<const LATrilinos::VectorType *> x_system (3);
+  x_system[0] = &y;
+  x_system[1] = &y_dot;
+  x_system[2] = &y_expl;
+
+
+  parallel::distributed::SolutionTransfer<dim, LATrilinos::VectorType, DoFHandler<dim,spacedim> >
+  system_trans (*dof_handler);
+
+  system_trans.prepare_serialization (x_system);
+
+  triangulation->save ((snap_prefix + "restart.mesh").c_str());
+}
```

Comment:
Why do you do this only for these vectors, and not for all of them, and some you save using block_write?